File: GenevaDebugging/GenevaLoggingLimit-DotNet/Tests/OtlpTest.cs
using System.Diagnostics;

namespace GenevaLoggingLimit_DotNet.Tests;

/// <summary>
/// Mirrors the C++ OtlpTest.cpp — entry points that wire up
/// the OTLP HTTP pipeline and run the shared test functions.
/// </summary>
internal static class OtlpTest
{
    private const string ServiceName = "MsaLab-Foo-Otlp";
    private const string TracerName = "Otlp-Tracer-Foo";
    private const string LoggerName = "Otlp-Logger-Foo";

    public static void TestTraceWithOtlp()
    {
        using var pipeline = OTelPipelineFactory.CreateOtlp(ServiceName);
        pipeline.Start();

        using var activitySource = new ActivitySource(TracerName);
        var logger = pipeline.LoggerFactory.CreateLogger(LoggerName);

        // TestRunner.TestTrace(logger, activitySource);

        pipeline.Shutdown();
    }

    public static void TestLogWithOtlp()
    {
        using var pipeline = OTelPipelineFactory.CreateOtlp(ServiceName);
        pipeline.Start();

        using var activitySource = new ActivitySource(TracerName);
        var logger = pipeline.LoggerFactory.CreateLogger(LoggerName);

        // TestRunner.TestLog(logger, activitySource);
        TestRunner.TestLogAttributeLimit(logger, activitySource);

        pipeline.Shutdown();
    }
}

File: otel/Geneva-ETW-Dll-Exe/GenevaEtwExe/test/OtlpTest.cpp

#include <map>
#include <string>
#include <vector>

#include "opentelemetry/trace/span_context.h"

#include "../IOtelPipeline.h"
#include "OtlpTest.h"
#include "test.h"

namespace {
  const std::string serviceName = "MsaLab-Foo-Otlp";
  const std::string tracerName = "Otlp-Tracer-Foo";
  const std::string loggerName = "Otlp-Logger-Foo";
}

namespace MsaLab { namespace Details
{
  const char* kGlobalProviderName = "OpenTelemetry-ETW-TLD-Geneva-Example";
  std::string providerName = kGlobalProviderName;

  using L = std::vector<std::pair<opentelemetry::trace::SpanContext, std::map<std::string, std::string>>>;

} // namespace Details
} // namespace MsaLab





